gaztec/src/utils/interactions.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  ContractArtifact,
+  FunctionArtifact,
+  encodeArguments,
+} from "@aztec/aztec.js";
+import { FunctionType } from "@aztec/foundation/abi";
```

Comment:
This should likely be included in aztec.js I'd imagine

Reply:
`aztec.js` is in dire need of a facelift, but no one even knew what was needed for an app like this until now, so it was minimalistic on purpose.